### ISSUE TITLE
drt_node  not delete 

### ISSUE THREAD

**Issue by chanchann** [matt-42/lithium#93]
https://github.com/matt-42/lithium/blob/master/libraries/http_server/http_server/dynamic_routing_table.hh

In line 48, you new drt_node, but I can't find the place you delete them.

```cpp
  else
  {
    auto new_node = new drt_node();
    children_.insert({k, new_node});
    return new_node->find_or_create(r, c);
  }
```

**Comment by spanasik** [comment]
leak reported for it for each registered handler called

```
==1362085==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 128 byte(s) in 1 object(s) allocated from:
    #0 0x7f001d4c0947 in operator new(unsigned long) (/lib/x86_64-linux-gnu/libasan.so.5+0x10f947)
    #1 0x55e97c23f1cc in li::internal::drt_node<li::api<li::http_request, li::http_response>::VH>::find_or_create(std::basic_string_view<char, std::char_traits<char> >, unsigned int) backend-cpp-l/lithium/single_headers/lithium_http_server.hh:3176
    #2 0x55e97c23f4ed in li::dynamic_routing_table<li::api<li::http_request, li::http_response>::VH>::operator[](std::basic_string_view<char, std::char_traits<char> > const&) backend-cpp-l/lithium/single_headers/lithium_http_server.hh:3245
    #3 0x55e97c23f658 in li::api<li::http_request, li::http_response>::operator()(int, std::basic_string_view<char, std::char_traits<char> >) backend-cpp-l/lithium/single_headers/lithium_http_server.hh:3475
    #4 0x55e97c237d5b in li::api<li::http_request, li::http_response>::get(std::basic_string_view<char, std::char_traits<char> >) backend-cpp-l/lithium/single_headers/lithium_http_server.hh:3480
    #5 0x55e97c21ed20 in main backend-cpp-l/my_api.cc:10
    #6 0x7f001cc550b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
```

**Comment by matt-42** [comment]
Hi  @spanasik thanks for the report. I fixed it in fef260d56a08ce83f7c042651b4f24e71db32025
